price-feeder/config/currency_provider_tracker.go

```diff
@@ -0,0 +1,175 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+const (
+	coinGeckoRestHost        = "https://api.coingecko.com/"
+	coinGeckoRestPath        = "api/v3/coins/"
+	coinGeckoListEndpoint    = "list"
+	coinGeckoTickersEndpoint = "tickers"
+	trackingPeriod           = time.Hour * 24
+)
+
+type (
+	// CurrencyProviderTracker queries the CoinGecko API for all the exchanges that
+	// support the currency pairs set in the price feeder config. It will poll the API
+	// every 24 hours to log any new exchanges that were added for a given currency.
+	//
+	// REF: https://www.coingecko.com/en/api/documentation
+	CurrencyProviderTracker struct {
+		logger              zerolog.Logger
+		pairs               []CurrencyPair
+		coinIDSymbolMap     map[string]string   // ex: map["ATOM"] = "cosmos"
+		CurrencyProviders   map[string][]string // map of price feeder currencies and what exchanges support them
+		CurrencyProviderMin map[string]int      // map of price feeder currencies and min required providers for them
+	}
+
+	// List of assets on CoinGecko and their corresponding id and symbol.
+	CoinGeckoCoinList struct {
+		ID     string `json:"id"`     // ex: "cosmos"
+		Symbol string `json:"symbol"` // ex: "ATOM"
+	}
+
+	// CoinGecko ticker shows market data for a given currency pair including what
+	// exchanges they're on.
+	CoinGeckoCoinTickerResponse struct {
+		Tickers []CoinGeckoCoinTicker `json:"tickers"`
+	}
+	CoinGeckoCoinTicker struct {
+		Base   string              `json:"base"`   // CurrencyPair.Base
+		Target string              `json:"target"` // CurrencyPair.Quote
+		Market CoinGeckoCoinMarket `json:"market"`
+	}
+	CoinGeckoCoinMarket struct {
+		Name string `json:"name"` // ex: Binance
+	}
+)
+
+func NewCurrencyProviderTracker(
+	ctx context.Context,
+	logger zerolog.Logger,
+	pairs ...CurrencyPair,
+) (*CurrencyProviderTracker, error) {
+	currencyProviderTracker := &CurrencyProviderTracker{
+		logger:              logger,
+		pairs:               pairs,
+		coinIDSymbolMap:     map[string]string{},
+		CurrencyProviders:   map[string][]string{},
+		CurrencyProviderMin: map[string]int{},
+	}
+
+	if err := currencyProviderTracker.setCoinIDSymbolMap(); err != nil {
+		return nil, err
+	}
+
+	if err := currencyProviderTracker.setCurrencyProviders(); err != nil {
+		return nil, err
+	}
+
+	currencyProviderTracker.setCurrencyProviderMin()
+
+	go currencyProviderTracker.trackCurrencyProviders(ctx)
+
+	return currencyProviderTracker, nil
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviders() map[string][]string {
+	return t.CurrencyProviders
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviderMin(base string) int {
+	return t.CurrencyProviderMin[base]
+}
```

Comment:
There's not really a point for having these getter functions without a mutex, would adding one be helpful for making sure we're threadsafe here?

Also, if we want to control how other packages are getting `CurrencyProviderMin` we should make it private